src/reference/90-Developers-Guide/02-Coding-Guideline/00.md

```diff
@@ -90,83 +90,87 @@ Use [MiMa][mima].
 
 #### Public traits should contain `def` declarations only
 
+TODO: "results to codegen at subclass" is unclear
+
 - `val` or `var` in `trait` results to codegen at subclass and at the artificial `Foo\$class.\$init\$`.
 - `lazy val` results to codegen at subclass
 
 #### Abstract classes are also useful
 
 [To trait, or not to trait?][pins_trait127].
-It is less flexible compared to trait, but easier to maintain binary compatibility. It also has better Java interop.
+Abtract classes are less flexible compared to traits, but ease maintaining binary compatibility. They also have better Java interoperability.
```

Comment:
I don't think it conveys the same meaning as the original, since you can't just add abstract classes on top of a trait to ease the pain.